runtime/gc/nogc/src/main/java/org/qbicc/runtime/gc/nogc/NoGcHelpers.java

```diff
@@ -1,44 +1,55 @@
 package org.qbicc.runtime.gc.nogc;
 
 import static org.qbicc.runtime.CNative.*;
-import static org.qbicc.runtime.posix.Stdlib.*;
-import static org.qbicc.runtime.stdc.Stddef.*;
-import static org.qbicc.runtime.stdc.Stdlib.*;
+import static org.qbicc.runtime.stdc.Stdint.*;
 import static org.qbicc.runtime.stdc.String.*;
 
 import org.qbicc.runtime.AutoQueued;
-import org.qbicc.runtime.Build;
 import org.qbicc.runtime.Hidden;
+import org.qbicc.runtime.gc.heap.Heap;
 
 /**
  *
  */
 public final class NoGcHelpers {
     private NoGcHelpers() {}
 
+    // our allocation position
+    private static long pos;
+
     @Hidden
     @AutoQueued
     public static Object allocate(long size, int align) {
```

Comment:
I was initially confused because if the current thread succeeds in calling `Heap.allocateRegion` but the `compareAndSetRelease` on `oldPos` fails we just restart the loop.  After reading more carefully, I convinced myself this is ok because the entire heap is being managed through a single bump pointer (`NoGc.pos`).  Therefore we didn't lose the region when we failed to commit the change from `oldPos` to `oldPos+size`....the heap is just growing linearly and when we try again that allocated region will be reflected in `Heap.getCurrentHeapOffset`, so it is not lost.